design-documents/graph-ql/coverage/dynamic-blocks.graphqls

```diff
@@ -0,0 +1,42 @@
+type DynamicBlock {
+    uid: ID!
+    content: String!
+}
+
+# We don't need name, is_enabled, locations, ga_creative, catalog_price_rule_ids, cart_price_rules_ids, locations on storefront
+
+type DynamicBlocksFilterInput {
+    type: DynamicBlockTypeEnum!
+    locations: [DynamicBlockLocationEnum!] # Blocks for all locations will be displayed if not supplied
+    rotation_mode: DynamicBlockRotationModeEnum!
+    dynamic_block_uids: [ID!] # Only makes sense when DynamicBlockTypeEnum is set to SPECIFIED
+}
+
+type DynamicBlocksOutput {
+    items: [DynamicBlock]!
+}
+
+enum DynamicBlockTypeEnum {
+    SPECIFIED
+    CART_PRICE_RULE_RELATED
+    CATALOG_PRICE_RULE_RELATED
+}
+
+enum DynamicBlockLocationEnum {
+    CONTENT
+    HEADER
+    FOOTER
+    LEFT
+    RIGHT
+}
```

Comment:
This is the kind of stuff I'd love to get ahead of in the future and design "Venia First". I'm unsure where we'd put a Dynamic Block that specified a LEFT or RIGHT location; it doesn't really fit.

No worries for now though.

Reply:
@supernova-at  it really comes to the question: is layout part of graphql or not.
since we have these zone is in Magento, relevant for Luma so far. Does it make sense to keep them for graphql.
Igor said yes, and I also got a request from @fnhipster to bring back the column layout on product interface. We took that one out completely

Reply:
Not sure what "Venia First" means but we need to cover data that might not be part of Venia but needed for other storefronts. Layout is a good example. It's not utilized in Venia (yet) but we use it to enable authors/admins to have multiple layouts.